consensus/src/proposal/block_generator.rs

```diff
@@ -104,12 +104,14 @@ impl<T: Operations> Generator<T> {
             result.txs.iter().map(|t| t.inner.hash()).collect();
         let txs: Vec<_> = result.txs.into_iter().map(|t| t.inner).collect();
         let txroot = merkle_root(&tx_hashes[..]);
+        let timestamp =
+            max(ru.timestamp() + MINIMUM_BLOCK_TIME, get_current_timestamp());
```

Comment:
Shouldn't the proposer wait instead of putting artificial timestamp that will be discarded?

Reply:
As we discussed, the artificial delay is done in the block acceptance but not in consensus. `get_current_timestamp()` will not be less than minimum block time.

Reply:
Both options have pros and cons. 

If we decide to go with `a delay prior to block propagation` then we need to increase `proposal step` default timeout (at least for iteration 0)

Reply:
We agreed on having the block generator create a block with a timestamp at least 10 seconds after the previous block. The candidate is broadcast immediately, but validators start validating candidates only if 10 seconds passed from the previous block. The generator (like all other provisioners) also waits a delay (if needed) to start collecting votes.